### PR TITLE
User profile page

### PR DESCRIPTION
**Pull request by ErikMichelson**
This PR adds the profile page with the following possiblilties:
- change user display name
- change password
- export user data
- delete user account

### CODE REVIEW

File: src/components/landing/pages/profile/settings/profile-change-password.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        <Card.Title><Trans i18nKey="profile.changePassword.title"/></Card.Title>
+        <Form onSubmit={updatePasswordSubmit} className="text-left">
+          <Form.Group controlId="oldPassword">
+            <Form.Label><Trans i18nKey="profile.changePassword.old"/></Form.Label>
+            <Form.Control
+              type="password"
+              size="sm"
+              className="bg-dark text-white"
+              required
+              value={oldPassword}
```
Comment (DerMolly) [inline src/components/landing/pages/profile/settings/profile-change-password.tsx:42]: this is not needed an leaks the password to the dom

File: src/components/landing/pages/profile/settings/profile-change-password.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            />
+          </Form.Group>
+          <Form.Group controlId="newPassword">
+            <Form.Label><Trans i18nKey="profile.changePassword.new"/></Form.Label>
+            <Form.Control
+              type="password"
+              size="sm"
+              className="bg-dark text-white"
+              required
+              value={newPassword}
```
Comment (DerMolly) [inline src/components/landing/pages/profile/settings/profile-change-password.tsx:53]: same problem

File: src/components/landing/pages/profile/settings/profile-change-password.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            <Form.Text><Trans i18nKey="profile.changePassword.info"/></Form.Text>
+          </Form.Group>
+          <Form.Group controlId="newPasswordAgain">
+            <Form.Label><Trans i18nKey="profile.changePassword.newAgain"/></Form.Label>
+            <Form.Control
+              type="password"
+              size="sm"
+              className="bg-dark text-white"
+              required
+              value={newPasswordAgain}
```
Comment (DerMolly) [inline src/components/landing/pages/profile/settings/profile-change-password.tsx:66]: see above

File: src/redux/user/types.ts [hunk 4]
```diff
@@ -20,11 +20,26 @@ export interface UserState {
...
+    github = 'github',
+    twitter = 'twitter',
+    gitlab = 'gitlab',
+    dropbox = 'dropbox',
+    google = 'google',
+    saml = 'saml',
+    oauth2 = 'oauth2',
+    email = 'email',
+    ldap = 'ldap',
+    openid = 'openid'
```
Comment (DerMolly) [inline src/redux/user/types.ts:42]: enums should be all caps, shouldn't they?

File: src/api/user.ts [hunk 5]
```diff
@@ -65,12 +65,13 @@ export const doDisplayNameUpdate: ((displayName: string) => Promise<void>) = asy
   expectResponseCode(response)
 }
 
-export const doPasswordChange: ((password: string) => Promise<void>) = async (password: string) => {
+export const doPasswordChange: ((oldPassword: string, newPassword: string) => Promise<void>) = async (oldPassword: string, newPassword: string) => {
```
Comment (DerMolly) [inline src/api/user.ts:68]:
```suggestion
export const doPasswordChange = async (oldPassword: string, newPassword: string): Promise<void> => {
```

File: src/components/landing/pages/profile/settings/profile-account-management.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
+  const [deletionButtonText, setDeletionButtonText] = useState('')
+  const [deletionButtonActive, setDeletionButtonActive] = useState(false)
+
+  const handleModalClose = () => {
+    setShowDeleteModal(false)
+  }
+
+  const doCountdown = (value: number) => {
+    if (value === 0) {
```
Comment (mrdrogdrog) [inline src/components/landing/pages/profile/settings/profile-account-management.tsx:20]: You should cancel the countdown if the modal was closed. If you open it, then close it and open it again, then two countdowns are batteling.

File: src/components/landing/pages/profile/settings/profile-change-password.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  useTranslation()
+  const [oldPassword, setOldPassword] = useState('')
+  const [newPassword, setNewPassword] = useState('')
+  const [newPasswordAgain, setNewPasswordAgain] = useState('')
+  const [newPasswordValid, setNewPasswordValid] = useState(false)
+  const [newPasswordAgainValid, setNewPasswordAgainValid] = useState(false)
+
+  const onChangeNewPassword = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewPassword(event.target.value)
+    setNewPasswordValid(/^[^\s].{5,}$/.test(event.target.value))
```
Comment (mrdrogdrog) [inline src/components/landing/pages/profile/settings/profile-change-password.tsx:16]: Please extract this regex into a const.

File: src/components/landing/pages/profile/settings/profile-display-name.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+export const ProfileDisplayName: React.FC = () => {
+  const { t } = useTranslation()
+  const user = useSelector((state: ApplicationState) => state.user)
+  const [submittable, setSubmittable] = useState(false)
+  const [error, setError] = useState(false)
+  const [displayName, setDisplayName] = useState(user.name)
+
+  const changeNameField = (event: ChangeEvent<HTMLInputElement>) => {
+    setSubmittable(!/^\s*$/.test(event.target.value))
```
Comment (mrdrogdrog) [inline src/components/landing/pages/profile/settings/profile-display-name.tsx:17]: Please extract this regex into a const

File: src/components/landing/pages/profile/settings/profile-display-name.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        <Card.Title>
+          <Trans i18nKey="profile.userProfile"/>
+        </Card.Title>
+        <Form onSubmit={changeNameSubmit} className="text-left">
+          <Form.Group controlId="displayName">
+            <Form.Label><Trans i18nKey="profile.displayName"/></Form.Label>
+            <Form.Control
+              type="text"
+              size="sm"
+              placeholder={t('displayName')}
```
Comment (mrdrogdrog) [inline src/components/landing/pages/profile/settings/profile-display-name.tsx:43]: wrong translation

**Pull request merged**